lnwire/dyn_begin_propose.go

```diff
@@ -0,0 +1,61 @@
+package lnwire
+
+import (
+	"bytes"
+	"io"
+)
+
+// BeginProposeFlags is the `begin_propose_flags` value sent in the
+// DynBeginPropose message during a dynamic commitment negotiation.
+type BeginProposeFlags uint8
+
+// DynBeginPropose is the message sent to initiate, as well as respond to the
+// initiation of, a dynamic commitment negotiation.
+type DynBeginPropose struct {
+	// ChanID identifies the channel for which we are trying to renegotiate
+	// the parameters
+	ChanID ChannelID
+
+	// Flags is a bit vector that contains a set of bits that one channel
+	// party can use to respond to a prior DynPropose message. Currently
+	// only the least significant bit is defined and is set when the sending
+	// party wishes to reject the terms offered in the prior received
+	// DynPropose message.
+	//
+	// TODO(proofofkeags): find a better way to do this...it was far too
+	// hard to explain why this structure is the way that it is.
+	Flags BeginProposeFlags
+}
+
+// A compile time check to ensure DynBeginPropose implements the lnwire.Message
+// interface.
+var _ Message = (*DynBeginPropose)(nil)
+
+// Encode serializes the target DynBeginPropose into the passed io.Writer.
+// Serialization will observe the rules defined by the passed protocol version.
+//
+// This is a part of the lnwire.Message interface.
+func (dbp *DynBeginPropose) Encode(w *bytes.Buffer, _ uint32) error {
+	if err := WriteChannelID(w, dbp.ChanID); err != nil {
```

Comment:
We should still read in the `ExtraOpaqueData` here. Otherwise we may end up with just a partially consumed buffer.

Reply:
I guess we also still need to decide which portions of the message are TLV vs otherwise.